### DISCUSSION TITLE
Push & Build Client Code

### DISCUSSION THREAD

**Discussion by gpathela** [aws-samples/aws-saas-factory-ref-solution-serverless-saas#58] [Q&A]
Hi

Just curious if there is any automated way to deploy client changes. At the moment if we push to CodeCommit it just builds server-side code and deploys only that.  I am manually building the code a pushing it to S3. Is there any automation possible here?

**Comment by ujwalbukka** [comment]
Hi

One way to automate this is, you can build and push client changes as a part of code build step.

Ideally, you should have a control plane repo which will have services shared across tenants like tenant-registration, tenant-management, admin UI..etc. An application plane repo which will have application services, application UI..etc. Then you should have a Codepipeline for each of these repos.
If you want to know more about control plane and application plane refer to this l[ink](https://docs.aws.amazon.com/whitepapers/latest/saas-architecture-fundamentals/control-plane-vs.-application-plane.html)
